ivy/src/main/scala/sbt/internal/librarymanagement/IvyActions.scala

```diff
@@ -497,8 +498,15 @@ object IvyActions {
       checkFilesPresent(artifacts)
       try {
         resolver.beginPublishTransaction(module.getModuleRevisionId(), overwrite);
-        for ((artifact, file) <- artifacts)
-          resolver.publish(artifact, file, overwrite)
+        artifacts.foreach {
+          case (artifact, file) =>
+            // should read max retries from some config?
+            IvyUtil.retryWithBackoff(
+              resolver.publish(artifact, file, overwrite),
+              TransientNetworkException.apply,
+              retryLimit = 5
+            )
```

Comment:
https://github.com/sbt/librarymanagement/blob/develop/core/src/main/scala/sbt/internal/librarymanagement/LMSysProp.scala is a place to keep system properties.
You could make something like

```scala
val maxPublishAttempts = java.lang.Integer.getInteger("sbt.repository.publish.retry", 3)
```

Reply:
Since we are counting attempts instead of retries, should we use property `sbt.repository.publish.attempts`?

Reply:
Yea. That sounds good.